examples_tests/48.OrientedBoundingBox/main.cpp

```diff
@@ -0,0 +1,153 @@
+
+        asset::SPushConstantRange pcRange = { asset::ISpecializedShader::ESS_VERTEX, 0u, sizeof(core::matrix4SIMD) };
+        auto pipelineLayout = driver->createGPUPipelineLayout(&pcRange, &pcRange + 1);
+
+        gpuPipeline = driver->createGPURenderpassIndependentPipeline(
+            nullptr, core::smart_refctd_ptr(pipelineLayout), 
+            shaders, shaders + 2u, 
+            mb->getPipeline()->getVertexInputParams(),
+            asset::SBlendParams(), asset::SPrimitiveAssemblyParams(), SRasterizationParams());
+    }
+}
+
+int main()
+{
+    // create device with full flexibility over creation parameters
+    // you can add more parameters if desired, check irr::SIrrlichtCreationParameters
+    irr::SIrrlichtCreationParameters params;
+    params.Bits = 24; //may have to set to 32bit for some platforms
+    params.ZBufferBits = 24; //we'd like 32bit here
+    params.DriverType = video::EDT_OPENGL; //! Only Well functioning driver, software renderer left for sake of 2D image drawing
+    params.WindowSize = dimension2d<uint32_t>(1280, 720);
+    params.Fullscreen = false;
+    params.Vsync = true; //! If supported by target platform
+    params.Doublebuffer = true;
+    params.Stencilbuffer = false; //! This will not even be a choice soon
+    auto device = createDeviceEx(params);
+
+    if (!device)
+        return 1; // could not create selected driver.
+
+
+    //! disable mouse cursor, since camera will force it to the middle
+    //! and we don't want a jittery cursor in the middle distracting us
+    device->getCursorControl()->setVisible(false);
+
+    //! Since our cursor will be enslaved, there will be no way to close the window
+    //! So we listen for the "Q" key being pressed and exit the application
+    QToQuitEventReceiver receiver;
+    device->setEventReceiver(&receiver);
+
+
+    auto* driver = device->getVideoDriver();
+    auto* smgr = device->getSceneManager();
+    auto* am = device->getAssetManager();
+    auto* fs = am->getFileSystem();
+
+    //load mesh
+    asset::IAssetLoader::SAssetLoadParams lp;
+    auto meshes_bundle = am->getAsset("../../media/cow.obj", lp);
+    assert(!meshes_bundle.isEmpty());
+    auto mesh = meshes_bundle.getContents().begin()[0];
+    auto mesh_raw = dynamic_cast<asset::ICPUMesh*>(mesh.get());
+    auto cpuMB = mesh_raw->getMeshBuffer(0u);
+
+    {
+        uint32_t posAttrIdx = cpuMB->getPositionAttributeIx();
+        uint8_t* posPtr = cpuMB->getAttribPointer(posAttrIdx);
+        const uint32_t stride = cpuMB->getPipeline()->getVertexInputParams().bindings[0].stride;
+        core::matrix3x4SIMD rs;
+        rs.setRotation(core::quaternion(0.0f, -3.1415f / 4.0f, -3.1415f / 4.0f));
+        core::matrix3x4SIMD s;
+        //s.setScale(core::vectorSIMDf(10.0f, 1.0f, 1.0f));
+        rs = core::matrix3x4SIMD::concatenateBFollowedByA(rs, s);
+
+        for (size_t i = 0ull; i < cpuMB->calcVertexCount(); i++)
```

Comment:
`calcVertexCount` is expensive, call once, not in a loop. Thats why it's called `calc`, not `get`